Signal/src/ViewControllers/ConversationView/MessagesViewController.m

```diff
@@ -2285,17 +2299,23 @@ - (void)updateLoadEarlierVisible
 
 - (void)updateMessageMappingRangeOptions
 {
-    // The "page" length may have been increased by loading "prev" pages at the
-    // top of the window.
-    NSUInteger pageLength = kYapDatabasePageSize * (self.page + 1);
-    // The "old" length may have been increased by insertions of new messages
+    // The "old" range length may have been increased by insertions of new messages
     // at the bottom of the window.
     NSUInteger oldLength = [self.messageMappings numberOfItemsInGroup:self.thread.uniqueId];
-    NSUInteger newLength = MAX(pageLength, oldLength);
+    // The "page-based" range length may have been increased by loading "prev" pages at the
+    // top of the window.
+    NSUInteger rangeLength;
+    while (YES) {
+        rangeLength = kYapDatabasePageSize * (self.page + 1);
+        if (rangeLength >= oldLength) {
+            break;
+        }
+        self.page = self.page + 1;
+    }
```

Comment:
That should read: A small fix to improve behavior of the "load more" button in conversations with a large amount of unread messages.